thirdparty/DiskANN/src/aux_utils.cpp

```diff
@@ -964,6 +968,13 @@ namespace diskann {
       }
     }
 
+    bool fast = false;
+    if (param_list.size() == 8){
+      if(1 == atoi(param_list[7].c_str())){
+        fast = true;
+      }
```

Comment:
line 953-965 of the original file also need to be taken care of.
Actually this param pass method is awful, since it brings a lot of troubles in param checking, reading, and using. If possible, can you help rephrase these param method by putting all params in the signature with default values?

Reply:
> line 953-965 of the original file also need to be taken care of. Actually this param pass method is awful, since it brings a lot of troubles in param checking, reading, and using. If possible, can you help rephrase these param method by putting all params in the signature with default values?

I am not sure how to improve it. Do we just modify the check params code in this function and not change any things, or also modify the params pass method which may involve UT and other files?

Reply:
We can modify the signature of `build_disk_index` by accepting all params. Only `IndexDiskANN` will get affected.
You can either do it here, or create another PR for it and merge that first.
Also the param check logic in `build_disk_index` also need to be changed.

Reply:
ok，I will modify it in this pr